src/rss_etl.py

```diff
@@ -0,0 +1,175 @@
+and loading it into Supabase.
+
+The database tables updated by this script are:
+    Bills
+    Revisions
+"""
+
+# from dotenv import load_dotenv
+# load_dotenv()  # for local secret management with .env file
+
+senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
+house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
+
+# sb_api_url = os.getenv("SUPABASE_API_URL")  # local secret management with .env file
+# sb_api_key = os.getenv("SUPABASE_API_KEY")  # local secret management with .env file
+sb_api_url = os.environ["SUPABASE_API_URL"]  # github actions secret management
+sb_api_key = os.environ["SUPABASE_API_KEY"]  # github actions secret management
+
+
+class LegislativeBody(Enum):
+    SENATE = "Senate"
+    HOUSE = "House"
+
+
+REVISIONS_OUTPUT_DICT = {
+    "bill_internal_id": "",
+    "printer_no": "",
+    "full_text_link": "",
+    "publication_date": "",
+    "revision_guid": "",
+    "description": "",
+}
+
+BILL_OUTPUT_DICT = {
+    "bill_number": "",
+    "legislative_session": "",
+    "session_type": "",
+    "chamber": "",
+    "legislative_id": "",
+}
+
+
+class BillIdentifier:
+    """
+    Extracts unique information from a bill's guid and embeds them in discrete properties
+    """
+
+    @staticmethod
+    def extract_bill_guid(revision_guid):
+        return re.search("BillGuid=(.*)", revision_guid).group(1)
+
+    @staticmethod
+    def determine_chamber(chamber_letter):
+        if chamber_letter == "H":
+            return "House"
+        elif chamber_letter == "S":
+            return "Senate"
+        else:
+            raise Exception("Invalid chamber letter in guid")
+
+    def __init__(self, guid):
+        self.guid = guid
+        self.bill_guid = re.search(r"(\d{4}\d[H|S]B\d+)P\d+", guid).group(1)
+        re_result = re.search(r"(\d{4})(\d)(H|S)B(\d+)P(\d+)", guid)
+        self.legislative_session = re_result.group(1)
+        self.session_type = re_result.group(2)
+        self.chamber = self.determine_chamber(re_result.group(3))
+        self.bill_number = re_result.group(4)
+        self.printer_number = re_result.group(5)
+
+
+class OutputRecord:
+
+    def __init__(self, supa_con, table_name, output_dict):
+        self.supa_con = supa_con
+        self.table_name = table_name
+        self.output_dict = output_dict.copy()
+
+    def has_existing_supabase_records(self, matching_columns):
+        # TODO: This might suffice for now, but selecting every row in the table is not a good long-term solution
+        statement = self.supa_con.table(self.table_name).select("*", count="exact")
+        for column in matching_columns:
+            statement = statement.eq(column, self.output_dict[column])
+        existing_bill_records = statement.execute()
+        return existing_bill_records.count > 0
+
+
+class Extractor:
+
+    def __init__(self, supa_con, leg_body, rss_feed):
+        self.supa_con = supa_con
+        self.leg_body = leg_body
```

Comment:
Nit:

Why not just call it "legistlative_body" instead of "leg_body", it's much clearer and the line won't be too long

Reply:
Good call -- added in next commit.